packages/block-editor/src/components/inserter/media-tab/utils.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress dependencies
+ */
+import { createBlock } from '@wordpress/blocks';
+
+const mediaTypeTag = { image: 'img', video: 'video', audio: 'audio' };
+
+export function getBlockAndPreviewFromMedia( media, mediaType ) {
+	// Add the common attributes between the different media types.
+	const attributes = {
+		id: media.id,
+	};
+	// Some props are named differently between the Media REST API and Media Library API.
+	// For example `source_url` is used in the former and `url` is used in the latter.
```

Comment:
Is this needed because this functions receives media from two places. Do you think it's possible for us to always use one source of media (say REST API)

Reply:
I don't think so, but this function will be revisited with Openverse integration, which is a different API. Ideally we found a good way to be able to add other sources as well.